File: PaymentServer/Result.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace PaymentServer
{
    struct Result<T>
    {
        bool success;
        T value;

        private Result(bool succes, T value)
        {
            this.success = succes;
            this.value = value;
        }
        public bool Success()
        {
            return success;
        }
        public T Value
        {
            get { return value; }
        }
        public static Result<T> Success(T value)
        {
            return new Result<T>(true, value);
        }
        public static Result<T> Failure(T value)
        {
            return new Result<T>(false, value);
        }
    }
}

File: PaymentServerJava/src/paymentserver/Result.java
package paymentserver;

public final class Result<T>
{
	private boolean success;
	private T value;

	public Result()
	{
	}

	private Result(boolean success, T value)
	{
		this.success = success;
		this.value = value;
	}
	public boolean Success()
	{
		return success;
	}
	public T getValue()
	{
		return value;
	}
	public static <T> Result<T> Success(T value)
	{
		return new Result<T>(true, value);
	}
	public static <T> Result<T> Failure(T value)
	{
		return new Result<T>(false, value);
	}

	public Result clone()
	{
		Result varCopy = new Result();

		varCopy.success = this.success;
		varCopy.value = this.value;

		return varCopy;
	}
}